bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/AbstractHomekitAccessoryImpl.java

```diff
@@ -209,8 +199,58 @@ protected void unsubscribe(HomekitCharacteristicType characteristicType) {
     protected <T> T getAccessoryConfiguration(@NonNull HomekitCharacteristicType characteristicType,
             @NonNull String key, @NonNull T defaultValue) {
         final Optional<HomekitTaggedItem> characteristic = getCharacteristic(characteristicType);
-        return characteristic.isPresent() ? getItemConfiguration(characteristic.get(), key, defaultValue)
-                : defaultValue;
+        return characteristic.isPresent() ? characteristic.get().getConfiguration(key, defaultValue) : defaultValue;
+    }
+
+    /**
+     * update mapping with values from item configuration.
+     * it checks for all keys from the mapping whether there is configuration at item with the same key and if yes,
+     * replace the value.
+     * 
+     * @param characteristicType characteristicType to identify item
+     * @param map mapping to update
+     */
+    protected void updateMapping(HomekitCharacteristicType characteristicType, Map<?, String> map) {
+        getCharacteristic(characteristicType).ifPresent(c -> {
+            final Map<String, Object> configuration = c.getConfiguration();
+            if (configuration != null) {
+                map.replaceAll((k, current_value) -> {
+                    final Object new_value = configuration.get(current_value);
+                    return (new_value instanceof String) ? (String) new_value : current_value;
+                });
+            }
+        });
+    }
+
+    /**
+     * takes item state as value and retrieves the key for that value from mapping.
+     * e.g. used to map StringItem value to HomeKit Enum
+     * 
+     * @param characteristicType characteristicType to identify item
+     * @param mapping mapping
+     * @param defaultValue default value if nothing found in mapping
+     * @param <T> type of the result derived from
+     * @return key for the value
+     */
+    protected <T extends java.lang.Enum<T>> T getKeyFromMapping(final HomekitCharacteristicType characteristicType,
```

Comment:
I don't know why you would need to force the generic type to an enum.
```suggestion
    protected <T> T getKeyFromMapping(final HomekitCharacteristicType characteristicType,
```
but if you really must.
```suggestion
    protected <T extends Enum<T>> T getKeyFromMapping(final HomekitCharacteristicType characteristicType,
```

Reply:
i was just thinking that Enum and EnumMap would help JVM optimise things.. but probably these few nano-seconds (if it at all) is not worth to make the method less generic. who knows, maybe we will need it for non-enum.